vision_datasets/common/data_manifest.py

```diff
@@ -499,6 +499,20 @@ def sample_few_shots_subset_greedy(self, num_min_samples_per_class, random_seed=
         sampled_images = [copy.deepcopy(x) for x in sampled_images]
         return DatasetManifest(sampled_images, self.labelmap, self.data_type)
 
+    def remove_images_without_labels(self):
+        """
+        Remove images without labels.
+        """
+        assert self.data_type in [DatasetTypes.IC_MULTICLASS, DatasetTypes.IC_MULTILABEL, DatasetTypes.OD]
+
+        sampled_images = []
+        images = list(self.images)
+        for image in images:
```

Comment:
images = [copy.deepcopy(image) for image in images if not self._is_negative(image.labels)]